drivers/SmartThings/matter-evse/src/init.lua

```diff
@@ -154,13 +183,33 @@ local EVSE_FAULT_STATE_ENUM_MAP = {
   [clusters.EnergyEvse.types.FaultStateEnum.OTHER] = capabilities.evseState.faultState.other,
 }
 
-local function read_cumulative_energy_imported(device)
-  local electrical_energy_meas_eps = embedded_cluster_utils.get_endpoints(device, clusters.ElectricalEnergyMeasurement.ID)
-  if electrical_energy_meas_eps and #electrical_energy_meas_eps > 0 then
-    local read_req = clusters.ElectricalEnergyMeasurement.attributes.CumulativeEnergyImported:read(device, electrical_energy_meas_eps[1])
-    for i, ep in ipairs(electrical_energy_meas_eps) do
+local BATTERY_CHARGING_STATE_MAP = {
+  [clusters.PowerSource.types.BatChargeStateEnum.IS_CHARGING] = capabilities.chargingState.chargingState.charging,
+  [clusters.PowerSource.types.BatChargeStateEnum.IS_NOT_CHARGING] = capabilities.chargingState.chargingState.stopped,
+  [clusters.PowerSource.types.BatChargeStateEnum.IS_AT_FULL_CHARGE] = capabilities.chargingState.chargingState.fullyCharged,
+}
+
+-- Matter Handlers
+local function read_cumulative_energy(device)
+  local cumul_eps = embedded_cluster_utils.get_endpoints(device,
+    clusters.ElectricalEnergyMeasurement.ID,
+    {feature_bitmap = clusters.ElectricalEnergyMeasurement.types.Feature.CUMULATIVE_ENERGY })
+  if cumul_eps and #cumul_eps > 0 then
+    local read_req = clusters.ElectricalEnergyMeasurement.attributes.CumulativeEnergyImported:read(device)
```

Comment:
Should there be a loop here similar to below in case there are multiple endpoints containing the ElectricalEnergyMeasurement cluster? Or is it not expected to have multiple endpoints?
```suggestion
    local read_req = clusters.ElectricalEnergyMeasurement.attributes.CumulativeEnergyImported:read(device)
    for i, ep in ipairs(cumul_eps) do
      if i > 1 then
        read_req:merge( clusters.ElectricalEnergyMeasurement.attributes.CumulativeEnergyExported:read(device, cumul_eps[i]))
      end
    end
```

Reply:
It is described as being included in only one endpoint when using EnergyImported in the specification. First of all, it would be nice to assume that there is only one and add it when there is a device that supports multiple EnergyImported.